lib/matplotlib/backend_bases.py

```diff
@@ -2253,9 +2253,21 @@ def get_default_filename(self):
         Return a string, which includes extension, suitable for use as
         a default filename.
         """
-        default_filename = self.get_window_title() or 'image'
-        default_filename = default_filename.lower().replace(' ', '_')
-        return default_filename + '.' + self.get_default_filetype()
+        default_basename = self.get_window_title() or 'image'
+        default_basename = default_basename.lower().replace(' ', '_')
+        default_filetype = self.get_default_filetype()
+        default_filename = default_basename + '.' + default_filetype
+
+        save_dir = os.path.expanduser(rcParams.get('savefig.directory', ''))
+
+        # ensure non-existing filename in save dir
+        i = 1
+        while os.path.isfile(os.path.join(save_dir, default_filename)):
+            # attach numerical value to basename
+            default_filename = '{}-{}.{}'.format(default_basename, i, default_filetype)
```

Comment:
Can't use this type of string formatting. It was introduced in py2.7 and we support py2.6. You will need "{0}-{1}.{2}" instead.